src/vm/types/signatures.rs

```diff
@@ -637,6 +644,24 @@ impl TypeSignature {
         )))
     }
 
+    pub fn max_string_ascii() -> TypeSignature {
+        SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(
+            BufferLength(
+                u32::try_from(MAX_VALUE_SIZE)
+                    .expect("FAIL: Max Clarity Value Size is no longer realizable in ASCII Type"),
+            ),
+        )))
+    }
+
+    pub fn max_string_utf8() -> TypeSignature {
+        SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(
+            StringUTF8Length(
+                u32::try_from(MAX_VALUE_SIZE)
```

Comment:
I think this should be `MAX_VALUE_SIZE / 4`, since Clarity UTF-8 strings represent each character as 4 bytes (even if the character itself can be represented in fewer bytes).

Reply:
OK. That makes sense to me. @kantai do you have an opinion on this?

Reply:
Hmm -- this actually raises a slightly separate but good point. What we want to do here is change this to something like:


```rust
    pub fn max_string_ascii() -> TypeSignature {
        SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(
            BufferLength::try_from(MAX_VALUE_SIZE)
                .expect("FAIL: Max Clarity Value Size is no longer realizable in ASCII Type"),
            ),
        ))
    }

    pub fn max_string_utf8() -> TypeSignature {
        SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(
            StringUTF8Length::try_from(MAX_VALUE_SIZE)
                .expect("FAIL: Max Clarity Value Size is no longer realizable in UTF8 Type"),
            ),
        ))
    }
```

Which will use the `try_from` check to enforce that the actual Clarity value size is legal (I believe @jcnelson is right that MAX_VALUE_SIZE isn't, but this way we'd get a panic indicating such).

Somewhat relatedly, we really should change `BufferLength(pub u32)` back to `BufferLength(u32)` -- I'll open a PR on `develop` that does exactly that. Nothing in the codebase violates the checked invariant, but the whole idea of the various `Length` classes is that their constructors enforce legal lengths.

Reply:
OK. Using StringUTF8Length::try_from(MAX_VALUE_SIZE / 4).